### PR TITLE
Fix reconnection failure when a connection remains in connected state

### PR DESCRIPTION
**Pull request by ryanjh**
e.g. SM/MAS/SCOB/BV-01-C

Signed-off-by: Chu, Ryan <ryan.chu@nordicsemi.no>

### CODE REVIEW

**Review by kromancer — APPROVED** [review]

**Review by AyturkDuzen — COMMENTED** [review]
I am still getting the failure on your branch, can we look at that together?

File: ptsprojects/zephyr/sm_wid.py [hunk 1]
```diff
@@ -16,6 +16,7 @@
 import logging
 import sys
 import re
+import time
```
Comment (kromancer) [inline ptsprojects/zephyr/sm_wid.py]: If we don't call `time.sleep` we won't need this one

**Review by kromancer — CHANGES_REQUESTED** [review]

File: ptsprojects/zephyr/sm_wid.py (hunk 1 above)
Comment (ryanjh) [reply ptsprojects/zephyr/sm_wid.py]: fixed

**Review by kromancer — APPROVED** [review]

**Comment by kromancer** [comment]
@AyturkDuzen I can reproduce the problem 9/10 times.
I am deeply confused on why I did not observe any problems when I was adding support for the test.
I am also not able to see any problems when running SM/SLA/SCOB/BV-02-C which is a "symmetric" test, the IUT is the slave/responder.

The failure manifests as a "Confirm Value Failed" coming from the lower tester (PTS) at the third round:
@ryanjh I strongly suggest that we open a ticket with the Bluetooth SIG and we merge this PR.

We can further discuss this issue in the upstream PR

**Comment by AyturkDuzen** [comment]
Not to block further progress, merged the commits. 
Left to do: @ryanjh creating SIG ticket with sniffer logs.

**Comment by ryanjh** [comment]
> The failure manifests as a "Confirm Value Failed" coming from the lower tester (PTS) at the third round:
> @ryanjh I strongly suggest that we open a ticket with the Bluetooth SIG and we merge this PR.

It is tracking by [CASE0070257](https://eur03.safelinks.protection.outlook.com/?url=https%3A%2F%2Fbluetooth.service-now.com%2Fess%2Fcase_detail.do%3Fsysparm_document_key%3Dx_bsig_case_manage_case%2Cfd3af7e41b0ce010186a4000cd4bcb0a&data=04%7C01%7CRyan.Chu%40nordicsemi.no%7C52a2c7a2f31d4ab2592108d875c315f3%7C28e5afa2bf6f419a8cf6b31c6e9e5e8d%7C0%7C0%7C637388828040049967%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=eUkX7JF04Ji98LqJX4Sd5qqX61tGf%2FVhrBHD7sOyupE%3D&reserved=0)

**Pull request merged**